plugins/s3-viewer-backend/src/credentials-provider/IAMRoleCredentialsProvider.ts

```diff
@@ -0,0 +1,82 @@
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { BucketCredentials, CredentialsProvider, S3Platform } from '../types';
+import { Config } from '@backstage/config';
+import { S3 } from '@aws-sdk/client-s3';
+
+export class IAMRoleCredentialsProvider implements CredentialsProvider {
```

Comment:
This should get a comment describing what it does and how it is different to the other providers.  Credentials are somehow (how?) received via an IAM role.

A link to official AWS docs about this would also be helpful.

Reply:
I added a small description and a link to the official documentation from AWS